internal/status/conditions.go

```diff
@@ -37,6 +37,7 @@ const ReasonDegraded RouteReasonType = "Degraded"
 const ReasonValid RouteReasonType = "Valid"
 const ReasonErrorsExist RouteReasonType = "ErrorsExist"
 const ReasonGatewayAllowMismatch RouteReasonType = "GatewayAllowMismatch"
+const ReasonRouteConflict RouteReasonType = "RouteConflict"
```

Comment:
Nit: I like that we have our own reasons, but we probably should look at pushing some of these standard ones upstream at some point.